application/src/main/java/org/opentripplanner/service/paging/PagingService.java

```diff
@@ -184,12 +186,33 @@ private PageCursorFactory mapIntoPageCursorFactory(@Nullable PageType currentPag
       searchWindowUsed
     );
 
+    handleFirstSearch(currentPageType, factory);
+
     if (pageCursorInput != null) {
       factory = factory.withRemovedItineraries(pageCursorInput);
     }
     return factory;
   }
 
+  /**
+   * If the first search is an arrive by search, add the latest itinerary's start time
+   * to allow setting cursor information correctly for the page cursor of the next page.
+   */
+  private void handleFirstSearch(@Nullable PageType currentPageType, PageCursorFactory factory) {
+    if (
+      currentPageType == null &&
+      PageCursorFactory.resolvePageTypeForTheFirstSearch(itinerariesSortOrder) == PREVIOUS_PAGE
+    ) {
+      if (itineraries.size() > 0) {
+        factory = factory.withFirstSearchLatestItineraryDeparture(
+          itineraries.get(0).startTimeAsInstant()
```

Comment:
So these are always sorted by the departure time, not by arrival time (i.e. if an itinerary starts later but arrives earlier than another one, is that first or second in the list)?